pkg/health/health_cert_manager.go

```diff
@@ -113,48 +176,84 @@ func GetCertificateHealth(obj *unstructured.Unstructured) (*HealthStatus, error)
 			continue
 		}
 
-		if c.Type == "Issuing" && cert.Status.NotBefore != nil {
+		if c.Type == certmanagerv1.CertificateConditionIssuing {
 			hs := &HealthStatus{
 				Status:  HealthStatusCode(c.Reason),
 				Ready:   false,
 				Message: c.Message,
 			}
 
-			if overdue := time.Since(cert.Status.NotBefore.Time); overdue > time.Hour {
-				hs.Health = HealthUnhealthy
-				return hs, nil
-			} else if overdue > time.Minute*15 {
-				hs.Health = HealthWarning
+			switch c.Reason {
```

Comment:
@adityathebe  Need to handle the pending case of a Issuer not found and on initial creation of certificate if age > period with no certificate status

Reply:
+ handle CertificateRequests in issueing state / pending state for too long